core/state.go

```diff
@@ -406,15 +417,18 @@ func calculateContractCommitment(storageRoot, classHash, nonce *felt.Felt) *felt
 	return crypto.Pedersen(crypto.Pedersen(crypto.Pedersen(classHash, storageRoot), nonce), &felt.Zero)
 }
 
-func (s *State) updateDeclaredClasses(declaredClasses []DeclaredV1Class) error {
+func (s *State) updateDeclaredClasses(declaredClasses []DeclaredV1Class, revert bool) error {
 	classesTrie, classesCloser, err := s.classesTrie()
 	if err != nil {
 		return err
 	}
 
 	for _, declaredClass := range declaredClasses {
 		// https://docs.starknet.io/documentation/starknet_versions/upcoming_versions/#commitment
-		leafValue := crypto.Poseidon(leafVersion, declaredClass.CompiledClassHash)
+		leafValue := &felt.Zero
+		if !revert {
+			leafValue = crypto.Poseidon(leafVersion, declaredClass.CompiledClassHash)
```

Comment:
Is setting the value to zero equivalent to deletion (in terms of taking up memory, querying the value of a key, etc)?